### PR TITLE
[web] Add TargetPlatform tests

### PR DESCRIPTION
**Pull request by ferhatb**
Adds target platform tests to be enabled when infra is ready.
These tests are for framework PR that will address issue : https://github.com/flutter/flutter/issues/43633

### CODE REVIEW

File: e2etests/web/regular_integration_tests/test_driver/target_platform_android_e2e.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import 'package:flutter_test/flutter_test.dart';
+import 'package:regular_integration_tests/text_editing_main.dart' as app;
+import 'package:flutter/material.dart';
+
+import 'package:e2e/e2e.dart';
+
+void main() {
+  E2EWidgetsFlutterBinding.ensureInitialized() as E2EWidgetsFlutterBinding;
+
+  testWidgets('Focused text field creates a native input element',
```
Comment (nturgut) [inline e2etests/web/regular_integration_tests/test_driver/target_platform_android_e2e.dart:16]: nit: change the method name

File: e2etests/web/regular_integration_tests/test_driver/target_platform_ios_e2e.dart [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import 'package:flutter_test/flutter_test.dart';
+import 'package:regular_integration_tests/text_editing_main.dart' as app;
+import 'package:flutter/material.dart';
+
+import 'package:e2e/e2e.dart';
+
+void main() {
+  E2EWidgetsFlutterBinding.ensureInitialized() as E2EWidgetsFlutterBinding;
+
+  testWidgets('Focused text field creates a native input element',
```
Comment (nturgut) [inline e2etests/web/regular_integration_tests/test_driver/target_platform_ios_e2e.dart:16]: nit: ditto

File: e2etests/web/regular_integration_tests/test_driver/target_platform_macos_e2e.dart [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import 'package:flutter_test/flutter_test.dart';
+import 'package:regular_integration_tests/text_editing_main.dart' as app;
+import 'package:flutter/material.dart';
+
+import 'package:e2e/e2e.dart';
+
+void main() {
+  E2EWidgetsFlutterBinding.ensureInitialized() as E2EWidgetsFlutterBinding;
+
+  testWidgets('Focused text field creates a native input element',
```
Comment (nturgut) [inline e2etests/web/regular_integration_tests/test_driver/target_platform_macos_e2e.dart:16]: nit: ditto

File: e2etests/web/regular_integration_tests/test_driver/target_platform_ios_e2e.dart (hunk 2 above)
Comment (ferhatb) [reply e2etests/web/regular_integration_tests/test_driver/target_platform_ios_e2e.dart]: Done.

File: e2etests/web/regular_integration_tests/test_driver/target_platform_macos_e2e.dart (hunk 3 above)
Comment (ferhatb) [reply e2etests/web/regular_integration_tests/test_driver/target_platform_macos_e2e.dart]: Done.

File: e2etests/web/regular_integration_tests/test_driver/target_platform_android_e2e.dart (hunk 1 above)
Comment (ferhatb) [reply e2etests/web/regular_integration_tests/test_driver/target_platform_android_e2e.dart]: Done.

**Pull request merged**